test/flask/snaps.spec.ts

```diff
@@ -41,13 +42,39 @@ describe.only("snaps", function () {
   });
 
   it("should install keys snap from npm", async function (this: TestContext) {
+    await installSnap(metamask.page, getSnapIdByName(this, Snaps.KEYS_SNAP), {
+      hasPermissions: true,
+      hasKeyPermissions: true,
+    });
+  });
+
+  it("should invoke provided snap method", async function (this: TestContext) {
+    const snapAddress = "http://localhost:8545";
     await installSnap(
       metamask.page,
-      "local:" + toUrl(this.snapServers[Snaps.KEYS_SNAP].address()),
+      getSnapIdByName(this, Snaps.PERMISSIONS_SNAP),
       {
         hasPermissions: true,
-        hasKeyPermissions: true,
-      }
+        hasKeyPermissions: false,
+      },
+      snapAddress
+    );
+
+    const testPage = await metamask.page.browser().newPage();
+    await testPage.goto(snapAddress);
+
+    const invokeAction = invokeSnap(
+      testPage,
+      getSnapIdByName(this, Snaps.PERMISSIONS_SNAP),
+      "hello",
+      { version: "latest" }
     );
+
+    await metamask.page.bringToFront();
+    await metamask.page.reload();
+
+    await clickOnButton(metamask.page, "Approve");
+
+    await invokeAction;
```

Comment:
assert expected result

Reply:
@mpetrunic thank you for the detailed review! I updated the code 😃